pkg/awsvpc/sg_analysis.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package awsvpc
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+
+	"github.com/np-guard/models/pkg/connection"
+	"github.com/np-guard/models/pkg/ipblock"
+	"github.com/np-guard/models/pkg/netp"
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/commonvpc"
+)
+
+type SpecificAnalyzer struct {
+	sgResource         *types.SecurityGroup
+	referencedIPblocks []*ipblock.IPBlock
+	sgMap              map[string]*commonvpc.SecurityGroup
+}
+
+func NewSpecificAnalyzer(sg *types.SecurityGroup) *SpecificAnalyzer {
+	res := &SpecificAnalyzer{sgResource: sg}
+	return res
+}
+
+func getAllConnSet() *connection.Set {
+	return connection.All()
+}
+
+func getProperty(p *int64, defaultP int64) int64 {
+	if p == nil {
+		return defaultP
+	}
+	return *p
+}
+
+func getTCPUDPConns(p string, srcPortMin, srcPortMax, dstPortMin, dstPortMax int64) *connection.Set {
```

Comment:
same comment as above - why is it here? there's nothing specific to `aws sg` about it.
Perhaps consider extracting all relevant functionality to another package so that it can be used by `ibmvpc, awsvpc` and `vpcmodel`

Reply:
moved

Reply:
I still see it in `ibmvpc`

Reply:
it is not there